File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution{
    public int longestConsecutive(int[] nums) {
        int maxLength = 0;
        HashMap<Integer, Boolean> map = new HashMap<>();
        for(int val : nums)
            map.put(val, true);
        
        for(int val : nums) 
            if(map.containsKey(val - 1))
                map.put(val, false);
        
        for(int val : nums) {
            if(map.get(val)) {
                int length = 1, start = val+1;
                
                while(map.containsKey(start))
                    start++;
                
                if((start-val)  > maxLength)
                    maxLength = start-val;
            }
        }
        
        return maxLength;     
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        var sequence = new Dictionary<int, bool>();
        foreach (int x in nums) sequence[x] = true;
        foreach (int x in nums) {
            if (sequence.ContainsKey(x - 1)) sequence[x] = false;
        }
        int max_len = 0;
        foreach (var x in sequence.Keys) {
            if (!sequence[x]) continue;
            int len = 1;
            int current = x;
            while (sequence.ContainsKey(++current)) len++;
            max_len = Math.Max(max_len, len);
        }
        
        return max_len;
    }
}
